staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -579,6 +579,8 @@ type APIResource struct {
 	// verbs is a list of supported kube verbs (this includes get, list, watch, create,
 	// update, patch, delete, deletecollection, and proxy)
 	Verbs Verbs `json:"verbs" protobuf:"bytes,4,opt,name=verbs"`
+	// shortName is the suggested short name of the resource.
+	ShortName string `json:"shortName, omitempty" protobuf:"bytes,5,opt,name=shortName"`
```

Comment:
@smarterclayton @lavalamp @liggitt this is the meat of the change right here.  This `shortName` would be used to allow `kubectl` to have short name representations of resources from aggregated API servers.  @p0lyn0mial already updated the shortcut expander to be group aware, so collisions won't be a problem.